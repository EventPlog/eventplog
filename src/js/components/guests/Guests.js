import React from 'react';
import styled from 'styled-components';

// internal components
import Tab from '../shared/tab';
import CheckInForm from './scenes/check-in-form/';
import GuestsList from './scenes/guests-list/';
import CsvImport from './scenes/csv-import'
import MessengerCheckIn from './scenes/messenger-check-in/';

const StyledGuests = styled.div`
`
const Guests = (props) => {
  const panes = [
    {name: 'All Guests', content: GuestsList},
    {name: 'Import CSV', content: CsvImport},
    {name: 'New guest', content: CheckInForm},
    {name: 'Messenger Code', content: MessengerCheckIn},
  ];

  return (
    <StyledGuests>
      <Tab panes={panes}/>
    </StyledGuests>
  )
}

export default Guests;
