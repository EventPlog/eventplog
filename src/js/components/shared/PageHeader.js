import React from 'react'
import { Header, Icon, Input, Segment } from 'semantic-ui-react'
import styled from 'styled-components';

const StyledPageHeader = styled.div`
  display: flex;
  justify-content: space-between;
  
  > h4.ui.header {
    margin: 0;
    padding-top: 10px;
    font-size: 1.07142857rem;
  }
`

const PageHeader = ({ title, icon }) => (
  <StyledPageHeader className="page-header">
    <Header as='h4'>
      <Icon name={icon || 'users'} />
      <Header.Content>
        {title}
      </Header.Content>
    </Header>
    <Input icon='search' placeholder='Search...' />
  </StyledPageHeader>
)

export default PageHeader;