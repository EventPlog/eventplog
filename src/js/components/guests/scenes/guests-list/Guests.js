import React from 'react'
import { Table, Icon, Menu } from 'semantic-ui-react'
import styled from 'styled-components';
import PageHeader from 'js/components/shared/PageHeader';
import Loading from 'js/components/shared/loading'
import Pagination from 'js/components/shared/pagination'
import GuestRow from './guest-row'

const StyledTable = styled.div`
  padding: 20px 0;
  
  .page-header {
    margin-bottom: 20px;
  }
  
`

const GuestsList = ({
  guests = {},
  getGuests,
}) => {
  const { data = [], meta = {}, loading, error }  = guests
  if (loading) return <Loading />
  if (error) return <Loading.Error msg={msg} />
  const startingIndex = meta.per_page * (meta.current_page - 1)
  return (
    <StyledTable>
      <PageHeader title="Guests" />
      <p>Total no. of guests: {meta && meta.total_count}</p>
      <Table celled unstackable>
        <Table.Header>
          <Table.Row>
            <Table.HeaderCell>S/N</Table.HeaderCell>
            <Table.HeaderCell>Full Name</Table.HeaderCell>
            <Table.HeaderCell>Checked In</Table.HeaderCell>
            <Table.HeaderCell>Feedback</Table.HeaderCell>
            <Table.HeaderCell> </Table.HeaderCell>
          </Table.Row>
        </Table.Header>

        <Table.Body>
          {data && data.map((guest, index) =>
            <GuestRow key={guest.id} {...{guest, index: (index + startingIndex)}} />
          )}
        </Table.Body>
        <Table.Footer>
          <Table.Row>
            <Table.HeaderCell colSpan='5'>
              <Menu floated='right' pagination>
                {
                  meta && meta.total_pages && data.length > 0
                    ? <Pagination totalPages={meta.total_pages}
                                  activePage={meta.current_page}
                                  onPageChange={getGuests} />
                    : ''
                }
              </Menu>
            </Table.HeaderCell>
          </Table.Row>
        </Table.Footer>
      </Table>
    </StyledTable>
  )
}

export default GuestsList