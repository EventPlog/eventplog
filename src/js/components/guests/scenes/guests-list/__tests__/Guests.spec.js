import React from 'react';
import Guests from '../Guests';
import { shallow } from 'enzyme';

import PageHeader from 'js/components/shared/PageHeader';

describe('Guests', () => {
  it('should render correctly', () => {
    const wrapper = shallow( <Guests  /> );

    expect(wrapper).toMatchSnapshot()
    expect(wrapper.find(PageHeader).length).toEqual(1);
  })
});