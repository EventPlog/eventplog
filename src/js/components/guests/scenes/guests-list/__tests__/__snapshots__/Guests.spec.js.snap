// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Guests should render correctly 1`] = `
<styled.div>
  <PageHeader
    title="Guests"
  />
  <p>
    Total no. of guests: 
  </p>
  <Table
    as="table"
    celled={true}
    unstackable={true}
  >
    <TableHeader
      as="thead"
    >
      <TableRow
        as="tr"
        cellAs="td"
      >
        <TableHeaderCell
          as="th"
        >
          S/N
        </TableHeaderCell>
        <TableHeaderCell
          as="th"
        >
          Full Name
        </TableHeaderCell>
        <TableHeaderCell
          as="th"
        >
          Checked In
        </TableHeaderCell>
        <TableHeaderCell
          as="th"
        >
          Feedback
        </TableHeaderCell>
        <TableHeaderCell
          as="th"
        >
           
        </TableHeaderCell>
      </TableRow>
    </TableHeader>
    <TableBody
      as="tbody"
    />
    <TableFooter
      as="tfoot"
    >
      <TableRow
        as="tr"
        cellAs="td"
      >
        <TableHeaderCell
          as="th"
          colSpan="5"
        >
          <Menu
            floated="right"
            pagination={true}
          />
        </TableHeaderCell>
      </TableRow>
    </TableFooter>
  </Table>
</styled.div>
`;
